### PR TITLE
UI navigation and customization improvements

### PR DESCRIPTION
**Pull request by gracenotes**
- Allow selecting and saving a custom font and font size
- When you cursor over a word, highlight all other instances of the word in the file (translucent box around the edges with customizable color)
- Ctrl+Click on a number to go to an event definition or event id, with basic regex search
- Make Japanese text non-overlapping in comments, although pretty small as a result
- Add compatibility mode for emevd files previously saved with an incorrect number of arguments
- Fully allow opening a JS file without a corresponding emevd file

### CODE REVIEW

File: DarkScript3/GUI.cs [hunk 1]
```diff
@@ -173,7 +174,7 @@ private void OpenToolStripMenuItem_Click(object sender, EventArgs e)
             }
             InstructionDocs oldDocs = Docs;
             OpenFileDialog ofd = new OpenFileDialog();
-            ofd.Filter = "EMEVD Files|*.emevd; *.emevd.dcx;";
+            ofd.Filter = "EMEVD Files|*.emevd; *.emevd.dcx|All files|*.*";
```
Comment (AinTunez) [inline DarkScript3/GUI.cs:177]: If we're going to explicitly allow opening JS files without a corresponding EMEVD, we should probably include `*.emevd.js` and `*.emevd.dcx.js` as a filter option.

File: DarkScript3/GUI.cs [hunk 2]
```diff
@@ -550,15 +680,15 @@ private void LoadColors()
             string[] lines = File.ReadAllLines("colors.cfg");
             foreach (string line in lines)
             {
-                string[] split = line.Split('=');
+                string[] split = line.Split(new[] { '=' }, 2);
                 string prop = split[0];
                 string val = split[1];
-                cfg[prop] = val;
+                cfg[prop] = val.Trim();
```
Comment (AinTunez) [inline DarkScript3/GUI.cs:686]: Why not trim the value?

File: DarkScript3/GUI.cs [hunk 3]
```diff
@@ -627,23 +806,37 @@ public static class JSRegex
...
+            public static Regex Number = new Regex(@"(\b|-)\d+[\.]?\d*([eE]\-?\d+)?[lLdDfF]?\b|\b0x[a-fA-F\d]+\b",
                                            RegexCompiledOption);
             public static Regex Keyword =
                 new Regex(
-                    @"\b(true|false|break|case|catch|const|continue|default|delete|do|else|export"
-                        + @"|for|function|if|in|instanceof|let|new|null|return|switch|this|throw|try|var|void|while|with|typeof)\b",
+                    @"\b("
+                        // Reserved in ECMAScript 2015
+                        + "break|case|catch|class|const|continue|debugger|default|delete|do|else|export|extends|finally|for|function"
+                        + "|if|import|in|instanceof|new|return|super|switch|this|throw|try|typeof|var|void|while|with|yield"
```
Comment (AinTunez) [inline DarkScript3/GUI.cs:817]: can we add `from` and `as`

**Review by AinTunez — CHANGES_REQUESTED** [review]

File: DarkScript3/GUI.cs (hunk 3 above)
Comment (gracenotes) [reply DarkScript3/GUI.cs:817]: I would be open to that, although they are not reserved words and still can be used as variable names. As discussed for now we can probably leave it off.

File: DarkScript3/GUI.cs (hunk 2 above)
Comment (gracenotes) [reply DarkScript3/GUI.cs]: I suppose I am trimming it, but I can trim the property name as well. This is mainly just a possible compatibility issue between \r and \r\n that wouldn't actually happen in practice, but may as well be more lenient.

File: DarkScript3/GUI.cs (hunk 1 above)
Comment (gracenotes) [reply DarkScript3/GUI.cs]: Done

**Pull request merged**